### ISSUE TITLE
R script for automatically generating random networks

### ISSUE THREAD

**Issue by kdahlquist** [kdahlquist/GRNmap#223]
@bklein7 has already implemented this; the code is posted on the DahlquistLab repository here:

https://github.com/kdahlquist/DahlquistLab/blob/master/R_scripts/Random-Matrix.R

From now on, I want to move discussion of this from e-mail to this issue.

I'm glad to know that the script is now working.  Here are some notes on the implementation:

* I wasn't actually able to get it to run.  I'm guessing that this might be because I don't know how to input the number of rows, columns, or connections.  I think R has the ability to generate a dialogue box to ask for the inputs, but in the meantime, I need to know where exactly to input these numbers.
* I think we should rename "connections" as "edges" because that is the nomenclature we are using for naming GRNmap files.
* I think the `setwd` command needs to be at the top of the script, instead of at the bottom to alert the user.  It probably needs to be uncommented and actually set to something.

**Comment by kdahlquist** [comment]
@bklein7 will look into `setwd` as a path relative to the directory that the script is in.

We need to inject some generic row and column headers.

**Comment by bklein7** [comment]
This week, I refined and expanded the functionality of this script. The following adjustments were made to the code:

* **Filler headings** were added to the rows and columns of the output CSV files.
  * The row and column headings will read "GENE_#".
  * Cell A1 now reads "rows affected/cols controlling" (this required a workaround).
* **Non-square matrices** are now properly generated by the script.
  * Genes only present in one group (either affected or controlling) will now be properly screened for connections to the network.
  * In the file name, the reported gene # will be that of the group (affecting/controlling) with the larger number of genes.
* The **nomenclature** used in the script was altered to match that of GRNmap.
  * File names are now output in the format: #-genes_#-edges_RandomNetwork (followed by a timestamp).
  * The function was renamed "random.network". Connection were renamed edges. References to row and column numbers now refer to the number of genes affecting and genes controlling, respectively.
* Added more extensive **comments** to the code.
  * For now, a sample setwd path was included at line 21, which the user can edit and uncomment before using the script.
  * Included a disclaimer explaining that GRNmap only accepts adjacency matrices.

**Comment by kdahlquist** [comment]
This all sounds good.  I'll give it a try as soon as I can--probably not until next Monday before the meeting.

**Comment by kdahlquist** [comment]
I have reviewed the script and it seems to function in the (limited) trials that I made.  I went ahead and made a few cosmetic changes to the code.
1. I changed the text in cell A1 to read "cols regulators/rows targets", which is the new convention
2. I made changes throughout the comments to indicate that the genes controlling are regulators and the genes affected are target genes.
3. I changed part of the output filename to read "random-network" instead of "RandomNetwork"
4. I wrote more of a step-by-step for how to run the function in RGui for Windows.
5. I updated the date to today to indicate a "Last modified" date.

I can think of ways to potentially break this code (by supplying non-integer values for the number of nodes and edges), but I think it does what we need it to do right now and we can move on.  We'll have  a quick discussion at the meeting before closing this.
